platform/fabric/core/generic/committer.go

```diff
@@ -125,29 +127,36 @@ func (c *Channel) commitUnknown(txID string, block uint64, indexInBlock int, env
 		return c.commitStoredEnvelope(txID, block, indexInBlock)
 	}
 
+	var envelopeRaw []byte
+	var err error
 	if envelope != nil {
 		// Store it
-		if err := c.EnvelopeService().StoreEnvelope(txID, envelope); err != nil {
+		envelopeRaw, err = proto.Marshal(envelope)
+		if err != nil {
 			return errors.WithMessagef(err, "failed to store unknown envelope for [%s]", txID)
 		}
 	} else {
 		// fetch envelope and store it
-		if err := c.FetchAndStoreEnvelope(txID); err != nil {
+		envelopeRaw, err = c.FetchEnvelope(txID)
+		if err != nil {
 			return errors.WithMessagef(err, "failed getting rwset for tx [%s]", txID)
 		}
 	}
 
 	// shall we commit this unknown envelope
-	if ok, err := c.filterUnknownEnvelope(txID); err != nil || !ok {
+	if ok, err := c.filterUnknownEnvelope(txID, envelopeRaw); err != nil || !ok {
```

Comment:
It doesn't make a big difference, but this method actually needs a `common.Envelope`, so we do a redundant unmarshaling. Then `StoreEnvelope` needs the byte array, so I would calculate both variables `envelope` and `envelopeRaw` in the if statements above. Then I'd pass `envelope` in filterUnknownEnvelope` and `envelopeRaw` in `StoreEnvelope`.